cli/src/commands/git/fetch.rs

```diff
@@ -54,52 +41,22 @@ pub fn cmd_git_fetch(
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
     let git_repo = get_git_repo(workspace_command.repo().store())?;
-    let remotes = if args.all_remotes {
-        get_all_remotes(&git_repo)?
-    } else if args.remotes.is_empty() {
-        get_default_fetch_remotes(ui, command.settings(), &git_repo)?
-    } else {
-        args.remotes.clone()
-    };
+
+    let remotes = get_fetch_remotes(ui, command.settings(), &git_repo, &args.fetch)?;
     let mut tx = workspace_command.start_transaction();
-    git_fetch(ui, &mut tx, &git_repo, &remotes, &args.branch)?;
+    git_fetch(
+        ui,
+        &mut tx,
+        &git_repo,
+        &FetchArgs {
+            branch: args.fetch.branch.clone(),
+            remotes: remotes.clone(),
+            all_remotes: args.fetch.all_remotes,
+        },
```

Comment:
Oh, we can't use the original `args.fetch`? That makes `args.fetch` useless..

Reply:
Maybe `FetchRemoteArgs` can be extracted instead of all-in `FetchArgs`, but I'm not pretty sure if that makes sense. Sorry for the noise.

Reply:
No worries at all, I may have jumped the gun on the Refactor before finalizing how it will be used. For `sync` it turns out we need to fetch all branches for all configured remotes[1], so the help text will be different after all, and I don't need to flatten them in. I've reverted that last change, back to just passing FetchArgs{} around with branch, remotes and all_remotes as the fields.


[1] - https://docs.google.com/document/d/1K4j1xQPFxtINhyOyKuCMTZ_iNrw8g-5_giBFMGoLjaI/edit?tab=t.0#bookmark=id.y91cr92o9czf

Reply:
Let's not add `branch` to the args struct then. It doesn't make sense to pass `branch` in to `get_fetch_remotes()`, and `all_remotes` to `git_fetch()`.

It might also be simpler to just make both `get_all_remotes()` and `get_default_fetch_remotes()` public.

Reply:
So I think I finally have something that looks descent:

* `git_fetch` takes `FetchArgs`; which only has fields for `branch` and `remotes`. This is useful since FetchArgs disambiguated the fields and makes it easier to know what goes where.
* `get_fetch_remotes` doesn't need `FetchArgs` and since `remotes` and `all_remotes` are different types completely, it doesn't look terrible if I just pass them in directly

wdyt?

Reply:
Seems okay.

FWIW, I wonder if `git_fetch()` can actually be reused in "git sync". Suppose "git sync" does rebase based on the stats returned by `git::fetch()`, "git sync" will have to do lower-level API.

Regarding `args.all_remotes`, copy-pasting `if args.all_remotes { ... }` block wouldn't be so bad.